### PR TITLE
pistol: init at 0.0.4

### PR DESCRIPTION
**Pull request by doronbehar**
###### Motivation for this change

Package my baby https://github.com/doronbehar/pistol

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/pistol/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  buildInputs = [
+    file
+  ];
+
+  meta = with stdenv.lib; {
+    description = "General purpose file previewer designed for Ranger, Lf to make scope.sh redundant";
+    homepage = "https://github.com/doronbehar/pistol";
+    license = licenses.mit;
+    maintainers = with maintainers; [ doronbehar ];
+    platforms = platforms.all;
```
Comment (jonringer) [inline pkgs/tools/misc/pistol/default.nix]: let the `buildGoModule` helper take care of this
```suggestion
```

**Review by jonringer — COMMENTED** [review]
otherwise diff LGTM

**Review by jonringer — COMMENTED** [review]
please squash commits

**Comment by doronbehar** [comment]
What say you @jonringer ?

**Review by jonringer — APPROVED** [review]
diff LGTM
commits LGTM
has tests :+1: 

```
[4 built, 0.0 MiB DL]
https://github.com/NixOS/nixpkgs/pull/80015
1 package built:
pistol
```

**Pull request merged**